docs/tsconfig.json

```diff
@@ -13,6 +13,5 @@
     "esModuleInterop": true,
     "incremental": true
   },
-  "include": ["next-env.d.ts", "next.config.ts", "types", "src", "pages", "data"],
```

Comment:
I spent quite some time wrapping my head around this issue, and I'm not quite sure why this helped 😅
This needs to be tested to make sure no TS issues are ignored by `docs:typescript:check`

Reply:
cc @Janpot

Reply:
Will check it out in detail tomorrow, but are you sure you didn't just turn off typechecking altogether by not including any file in the compilation?

Reply:
> are you sure you didn't just turn off typechecking altogether by not including any file in the compilation?

The default value for `include` is

> `[]` if [files](https://www.typescriptlang.org/tsconfig/#files) is specified; `**/*` otherwise.
> https://www.typescriptlang.org/tsconfig/#include

We don't have `files` specified, so IIUC `**/*` matches all files except those that are excluded.

> We reproduced the issue in isolation in the X repo, so I doubt this is fixing it.

I wasn't aware of this. Can you point me to it?
If that's the case, I must agree that my changes here are pointless 🙃

Reply:
> I wasn't aware of this. Can you point me to it?

This is the PR, https://github.com/mui/mui-x/pull/17612
It has been altered, the original reproduction is https://github.com/mui/mui-x/pull/17612/commits/0ba6be6bb3557822defddeaa3ed244d20072d048

Have a messy PR where I'm bisecting the core docs as well https://github.com/mui/material-ui/pull/46043

Reply:
Ongoing conversation [here](https://mui-org.slack.com/archives/C02EQ97NV8F/p1745511207073429)